src/utils/useAnchorProvider.ts

```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react'
```

Comment:
I wonder if it would be worth adding this to the global context, so we don't have to recreate it?

Reply:
Hmm yeah seems worth it. Let me make that change